tests/integration/tls_tests/test_tls.py

```diff
@@ -36,6 +46,69 @@ async def test_enable_tls(ops_test: OpsTest) -> None:
 
 
+async def test_set_tls_key(ops_test: OpsTest) -> None:
+    """Verify rotating tls private keys restarts mongod with new certificates."""
+    # dict of values for cert file certion and mongod service start times. After resetting the
+    # private keys these certificates should be updated and the mongod service should be
+    # restarted
+    original_tls_times = {}
+    for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
+        original_tls_times[unit.name] = {}
+        original_tls_times[unit.name]["external_cert"] = await time_file_created(
+            ops_test, unit.name, EXTERNAL_CERT_PATH
+        )
+        original_tls_times[unit.name]["internal_cert"] = await time_file_created(
+            ops_test, unit.name, INTERNAL_CERT_PATH
+        )
+        original_tls_times[unit.name]["mongod_service"] = await time_process_started(
+            ops_test, unit.name, DB_SERVICE
+        )
+
+    with open(PRIVATE_KEY_PATH) as f:
+        key_contents = f.readlines()
+        key_contents = "".join(key_contents)
+
+    # set internal key
+    leader_unit = await find_unit(ops_test, leader=True)
+    leader_unit = await find_unit(ops_test, leader=True)
+    await leader_unit.run_action(
+        action_name="set-tls-private-key", params="=".join(["external-key", key_contents])
+    )
+
+    # set external key
+    leader_unit = await find_unit(ops_test, leader=True)
+    await leader_unit.run_action(
+        action_name="set-tls-private-key", params="=".join(["internal-key", key_contents])
+    )
+
+    # sleep for long enough to process new certificate request and update certs
+    sleep(30)
+
+    # After updating both the external key and the internal key a new certificate request will be
+    # made; then the certificates should be available and updated.
+    for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
+        new_external_cert_time = await time_file_created(ops_test, unit.name, EXTERNAL_CERT_PATH)
+        new_internal_cert_time = await time_file_created(ops_test, unit.name, INTERNAL_CERT_PATH)
+        new_mongod_service_time = await time_process_started(ops_test, unit.name, DB_SERVICE)
+
+        assert (
+            new_external_cert_time > original_tls_times[unit.name]["external_cert"]
+        ), f"external cert for {unit.name} was not updated."
+        assert (
+            new_internal_cert_time > original_tls_times[unit.name]["internal_cert"]
+        ), f"internal cert for {unit.name} was not updated."
+
+        # Once the certificate requests are processed and updated the mongod.service should be
+        # restarted
+        assert (
+            new_mongod_service_time > original_tls_times[unit.name]["mongod_service"]
+        ), f"mongod service for {unit.name} was not restarted."
+
+    # Verify that TLS is functioning on all units.
+    for unit in ops_test.model.applications[DATABASE_APP_NAME].units:
+        assert await check_tls(ops_test, unit, enabled=True)
```

Comment:
may be to confirm, it'd be nice to check that tls is disabled before setting the certificates?

Reply:
TLS is already enabled at this point (in the previous test) so it shouldn't be disabled before setting the cert.

This check verifies that resetting it didn't break anything, i.e. TLS is still enabled. By verifying the service restarted (above) and the new files were created (also above) this verifies that it we infact changed TLS cert and restarted mongod with the new TLS cert and that this new cert works